src/main/java/eu/europa/ted/eforms/sdk/selector/component/VersionDependentComponentFactory.java

```diff
@@ -1,74 +1,105 @@
+import org.reflections.util.ConfigurationBuilder;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public abstract class VersionDependentComponentFactory {
-  private static final Logger logger = LoggerFactory.getLogger(VersionDependentComponentFactory.class);
+  private static final Logger logger =
+      LoggerFactory.getLogger(VersionDependentComponentFactory.class);
 
   private Map<String, Map<VersionDependentComponentType, VersionDependentComponentDescriptor<?>>> componentsMap;
 
-  protected VersionDependentComponentFactory() {
-    populateComponents();
+  protected static final String[] DEFAULT_PACKAGES = new String[] {"eu.europa.ted"};
+
+  /**
+   * A factory for instantiating classes annotated with {@link VersionDependentComponent}. 
+   * @param packages The package(s) to scan for annotated classes
+   */
+  protected VersionDependentComponentFactory(String... packages) {
+    populateComponents(Optional.ofNullable(packages).orElse(DEFAULT_PACKAGES));
   }
 
-  private void populateComponents() {
-    if (componentsMap == null) {
-      componentsMap = new HashMap<>();
+  private void populateComponents(String... packages) {
+    if (ArrayUtils.isEmpty(packages)) {
+      packages = DEFAULT_PACKAGES;
     }
 
-    ClassIndex.getAnnotated(VersionDependentComponent.class).forEach((Class<?> clazz) -> {
-      VersionDependentComponent annotation = clazz.getAnnotation(VersionDependentComponent.class);
-
-      String[] supportedSdkVersions = annotation.versions();
-      VersionDependentComponentType componentType = annotation.componentType();
-
-      for (String sdkVersion : supportedSdkVersions) {
-        Map<VersionDependentComponentType, VersionDependentComponentDescriptor<?>> components =
-            componentsMap.get(sdkVersion);
+    Class<VersionDependentComponent> annotationType = VersionDependentComponent.class;
 
-        if (components == null) {
-          components = new HashMap<>();
-          componentsMap.put(sdkVersion, components);
-        }
+    logger.debug("Looking in the classpath for types annotated with {}", annotationType);
 
-        VersionDependentComponentDescriptor<?> component =
-            new VersionDependentComponentDescriptor<>(sdkVersion, componentType, clazz);
-        VersionDependentComponentDescriptor<?> existingComponent = components.get(componentType);
-
-        if (existingComponent != null && !existingComponent.equals(component)) {
-          throw new IllegalArgumentException(MessageFormat.format(
-              "More than one components of type [{0}] have been found for SDK version [{1}]:\n\t- {2}\n\t- {3}",
-              componentType, sdkVersion, existingComponent.getImplType().getName(),
-              clazz.getName()));
-        }
+    if (componentsMap == null) {
+      componentsMap = new HashMap<>();
+    }
 
-        components.put(componentType, component);
-      }
-    });
+    new Reflections(ConfigurationBuilder.build().forPackages(packages))
```

Comment:
All of this looks quite interesting but I sadly have no time to look into the details.